doc/source/reference/arrays.promotion.rst

```diff
@@ -0,0 +1,263 @@
+precision, this leads to the 
+
+
+Notable or special promotion behaviors
+--------------------------------------
+
+In NumPy promotion refers to what specific functions do with the result and
+in some cases, this means that NumPy may deviate from what the `np.result_type`
+would give.
+
+Behavior of ``sum`` and ``prod``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+**``np.sum`` and ``np.prod``:** Will alway return the default integer type
+when summing over integer values (or booleans).  This is usually an ``int64``.
+The reason for this is that integer summations are otherwise very likely
+to overflow and give confusing results.
+This rule also applies to the underlying ``np.add.reduce``, ``np.multiply.reduce``
+and other reduction methods.
+
+Notable behavior with NumPy or Python integer scalars
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+NumPy promotion refers to the result dtype and operation precision,
+but the operation will sometimes dictate that result.
+Division always returns floating point values and comparison always booleans.
+
+This leads to some special behaviors which may appear as "exceptions" to
+the rules:
+* NumPy comparisons with Python integers or mixed precision integers always
+  return the correct result.  The inputs will never be cast in a way which
+  loses precision.
+* Equality comparisons between types which cannot be promoted will be
+  considered all ``False`` (equality) or all ``True`` (not-equal).
+* Unary math functions like ``np.sin`` that always return floating point
+  values, accept any Python integer input by converting it to ``float64``.
+* Division always returns floating point values and thus also allows divisions
+  between any NumPy integer with any Python integer value by casting both
+  to ``float64``.
+
+It should be noted that while these exceptions can apply to more functions,
+NumPy may choose to not implement them in all such cases.
+
+Promotion of non-numerical datatypes
+------------------------------------
+
+NumPy extends the promotion to non-numerical types, although in most cases
+promotion is not well defined and simply rejected.
+
+The following rules apply:
+* NumPy byte strings (``np.bytes_``) can be promoted to unicode strings
+  (``np.str_``).  A ``result_type``
+* For some purposes NumPy will promote almost any other datatype to strings.
+  This applies to array creation or concatenation.
+* The array constructers like ``np.array()`` will use ``object`` dtype when
+  there is no viable promotion.
+* Structured dtypes can promote when their field names and order matches.
+  In that case all fields are promoted individually.
+* NumPy ``timedelta`` can in some cases promote with integers.
+
+.. note::
+    Some of these rules describe the state as of NumPy 2.0 and details
+    may be surprising and are under consideration to be change in the future.
+    However, changes always have to be weighed against backwards compatibility
+    concerns.
```

Comment:
Add: "Please raise an issue if you have particular ideas about how promotion should work"